spikeinterface/preprocessing/zero_channel_pad.py

```diff
@@ -0,0 +1,87 @@
+from typing import Union
+import numpy as np
+from spikeinterface.core import BaseRecording, BaseRecordingSegment
+from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
+from spikeinterface.core.core_tools import define_function_from_class
+
+
+class ZeroChannelPaddedRecording(BaseRecording):
+    name = 'zero_channel_pad'
+    installed = True
+
+    def __init__(self, parent_recording: BaseRecording,
+                 num_channels: int, channel_mapping: Union[list, None] = None):
+        """Pads a recording with channels that contain only zero.
+
+        Parameters
+        ----------
+        parent_recording : BaseRecording
+            recording to zero-pad
+        num_channels : int
+            Total number of channels in the zero-channel-padded recording
+        channel_mapping : Union[list, None], optional
+            Mapping from the channel index in the original recording to the zero-channel-padded recording, 
+            by default None.
+            If None, sorts the channel indices in ascending y channel location and puts them at the 
+            beginning of the zero-channel-padded recording.
+        """
+        BaseRecording.__init__(self, parent_recording.get_sampling_frequency(), 
+                               np.arange(num_channels), parent_recording.get_dtype())
+        
+        if channel_mapping is not None:
+            assert len(channel_mapping) == parent_recording.get_num_channels(), \
+                "The new mapping must be specified for all channels."
+            assert max(channel_mapping) < num_channels, ("The new mapping cannot exceed total number of channels "
+                                                         "in the zero-chanenl-padded recording.")
+        else:
+            if 'locations' in parent_recording.get_property_keys() or \
+                'contact_vector' in parent_recording.get_property_keys():
+                self.channel_mapping = np.argsort(parent_recording.get_channel_locations()[:, 1])
```

Comment:
This is dangerous because the if 2 contacts on the same depth the order is random.
And maybe this dim `1` should be controlled by an option

Reply:
We always assume the second dimension is depth though..also in plotting and localization